packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart

```diff
@@ -116,4 +123,65 @@ void main() {
               message: 'It appears the output file contains an error message, not valid output.'));
     });
   });
+
+  group('Screenshot for devices unsupported for project', () {
+    late _TestDeviceManager testDeviceManager;
+
+    setUp(() {
+      testDeviceManager = _TestDeviceManager(logger: BufferLogger.test());
+    });
+
+    testUsingContext('should not throw', () async {
+      final ScreenshotCommand command = ScreenshotCommand(fs: MemoryFileSystem.test());
+
+      final _ScreenshotDevice deviceUnsupportedForProject = _ScreenshotDevice(isSupportedForProject: false);
+
+      testDeviceManager.devices = <Device>[deviceUnsupportedForProject];
+
+      expect(() => createTestCommandRunner(command).run(<String>['screenshot']), returnsNormally);
```

Comment:
I don't think `returnsNormally` is an async matcher, and thus it will not catch if the command fails asynchronously. why don't you just await the call, without an expects around it.

Reply:
When I await the call without the `expect`, and I use `findTargetDevice(includeDevicesUnsupportedByProject: false)` in line 81 to make the call fail, the test fails but does not complete, and I have to `Cntrl-C` to exit it. 

```dart
    testUsingContext('should not throw for a single device', () async {
      final ScreenshotCommand command = ScreenshotCommand(fs: MemoryFileSystem.test());

      final _ScreenshotDevice deviceUnsupportedForProject = _ScreenshotDevice(isSupportedForProject: false);

      testDeviceManager.devices = <Device>[deviceUnsupportedForProject];

      await createTestCommandRunner(command).run(<String>['screenshot']);
    }, overrides: <Type, Generator>{
      DeviceManager: () => testDeviceManager,
    });
```

<details>
<summary>Logs:</summary>

_Test process is stuck after this:_

```bash
00:00 +8: Screenshot for devices unsupported for project should not throw for a single device
UnimplementedError: targetPlatform
#0      Fake.noSuchMethod (package:test_api/src/frontend/fake.dart:50:5)
#1      _ScreenshotDevice.targetPlatform (file:///Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart:160:7)
#2      Device.descriptions (package:flutter_tools/src/device.dart:821:58)
#3      TargetDevices._printUnsupportedDevice (package:flutter_tools/src/runner/target_devices.dart:333:23)
#4      TargetDevices._handleNoDevices (package:flutter_tools/src/runner/target_devices.dart:219:11)
<asynchronous suspension>
#5      FlutterCommand.findTargetDevice (package:flutter_tools/src/runner/flutter_command.dart:1766:32)
<asynchronous suspension>
#6      ScreenshotCommand._validateOptions (package:flutter_tools/src/commands/screenshot.dart:81:18)
<asynchronous suspension>
#7      ScreenshotCommand.verifyThenRunCommand (package:flutter_tools/src/commands/screenshot.dart:100:5)
<asynchronous suspension>
#8      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1374:27)
<asynchronous suspension>
#9      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#10     CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#11     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:349:9)
<asynchronous suspension>
#12     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#13     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:294:5)
<asynchronous suspension>
#14     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#15     main.<anonymous closure>.<anonymous closure> (file:///Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart:141:7)
<asynchronous suspension>
#16     testUsingContext.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (file:///Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/src/context.dart:141:26)
<asynchronous suspension>
#17     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#18     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#19     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
<asynchronous suspension>
#20     testUsingContext.<anonymous closure> (file:///Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/src/context.dart:102:5)
<asynchronous suspension>
#21     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:215:9)
<asynchronous suspension>
#22     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:213:7)
<asynchronous suspension>
#23     Invoker._waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:258:9)
<asynchronous suspension>

00:00 +8 -1: Screenshot for devices unsupported for project should not throw for a single device [E]
  UnimplementedError: targetPlatform
  package:test_api                                                 Fake.noSuchMethod
  test/commands.shard/hermetic/screenshot_command_test.dart 160:7  _ScreenshotDevice.targetPlatform
  package:flutter_tools/src/device.dart 821:58                     Device.descriptions
  package:flutter_tools/src/runner/target_devices.dart 333:23      TargetDevices._printUnsupportedDevice
  package:flutter_tools/src/runner/target_devices.dart 219:11      TargetDevices._handleNoDevices
  ===== asynchronous gap ===========================
  dart:async                                                       _CustomZone.registerBinaryCallback
  package:flutter_tools/src/base/context.dart 150:19               AppContext.run.<fn>
  dart:async                                                       runZoned
  package:flutter_tools/src/base/context.dart 149:12               AppContext.run
  test/src/context.dart 130:30                                     testUsingContext.<fn>.<fn>.<fn>.<fn>
  dart:async                                                       runZonedGuarded
  test/src/context.dart 128:18                                     testUsingContext.<fn>.<fn>.<fn>
  package:flutter_tools/src/base/context.dart 150:29               AppContext.run.<fn>
  dart:async                                                       runZoned
  package:flutter_tools/src/base/context.dart 149:12               AppContext.run
  test/src/context.dart 103:22                                     testUsingContext.<fn>.<fn>
  package:flutter_tools/src/context_runner.dart 84:18              runInContext.runnerWrapper
  ===== asynchronous gap ===========================
  dart:async                                                       _CustomZone.registerUnaryCallback
  package:flutter_tools/src/context_runner.dart 83:20              runInContext.runnerWrapper
  package:flutter_tools/src/base/context.dart 150:29               AppContext.run.<fn>
  dart:async                                                       runZoned
  package:flutter_tools/src/base/context.dart 149:12               AppContext.run
  package:flutter_tools/src/context_runner.dart 87:18              runInContext
  test/src/context.dart 102:11                                     testUsingContext.<fn>
  test/src/common.dart 183:18                                      test.<fn>
```

_After `Ctrl-C`:_

```bash

^C
To run this test again: /Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/bin/cache/dart-sdk/bin/dart test /Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart -p vm --plain-name 'Screenshot for devices unsupported for project should not throw for a single device'
unhandled error during finalization of test:
/Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/herme
tic/screenshot_command_test.dart
TestDeviceException(Shell subprocess terminated by ^C (SIGINT, -2).)
#0      FlutterTesterTestDevice.finished (package:flutter_tools/src/test/flutter_tester_device.dart:241:5)
<asynchronous suspension>
#1      FlutterTesterTestDevice.kill (package:flutter_tools/src/test/flutter_tester_device.dart:223:5)
<asynchronous suspension>
#2      FlutterPlatform._startTest.<anonymous closure> (package:flutter_tools/src/test/flutter_platform.dart:510:9)
<asynchronous suspension>
#3      FlutterPlatform._startTest (package:flutter_tools/src/test/flutter_platform.dart:566:11)
<asynchronous suspension>


unhandled error during finalization of test:
/Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/herme
tic/screenshot_command_test.dart
PathNotFoundException: Deletion failed, path = '/var/folders/8n/qpsf1kw9371gl1_vvgt4_05r0000gn/T/flutter_tools.rcRMz1/flutter_test_listener.dbAeH2'
(OS Error: No such file or directory, errno = 2)
#0      _Directory._deleteSync (dart:io/directory_impl.dart:193:7)
#1      FileSystemEntity.deleteSync (dart:io/file_system_entity.dart:423:7)
#2      ForwardingFileSystemEntity.deleteSync (package:file/src/forwarding/forwarding_file_system_entity.dart:70:16)
#3      FlutterPlatform._createListenerDart.<anonymous closure> (package:flutter_tools/src/test/flutter_platform.dart:602:15)
#4      FlutterPlatform._startTest (package:flutter_tools/src/test/flutter_platform.dart:566:26)
<asynchronous suspension>


02:39 +8 -2: loading /Users/victoreronmosele/development/mobile_projects/flutter_projects/flutter_contributions/flutter/packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart [E]
  Error: the Dart compiler exited unexpectedly.
  package:flutter_tools/src/base/common.dart 10:3  throwToolExit
  package:flutter_tools/src/compile.dart 896:9     DefaultResidentCompiler._compile.<fn>
  dart:async/zone.dart 1407:47                     _rootRunUnary
  dart:async/zone.dart 1308:19                     _CustomZone.runUnary
  dart:async/future_impl.dart 838:45               Future._propagateToListeners.handleValueCallback
  dart:async/future_impl.dart 867:13               Future._propagateToListeners
  dart:async/future_impl.dart 643:5                Future._completeWithValue
  dart:async/future_impl.dart 713:7                Future._asyncCompleteWithValue.<fn>
  dart:async/zone.dart 1399:13                     _rootRun
  dart:async/zone.dart 1301:19                     _CustomZone.run
  dart:async/zone.dart 1209:7                      _CustomZone.runGuarded
  dart:async/zone.dart 1249:23                     _CustomZone.bindCallbackGuarded.<fn>
  dart:async/schedule_microtask.dart 40:21         _microtaskLoop
  dart:async/schedule_microtask.dart 49:5          _startMicrotaskLoop
  dart:isolate-patch/isolate_patch.dart 118:13     _runPendingImmediateCallback
  dart:isolate-patch/isolate_patch.dart 185:5      _RawReceivePort._handleMessage
```

</details>

Reply:
This is an unrelated tool bug, any test failure will hang the process and require killing the process.

Reply:
Okay. I'll proceed with the change to use `await` then.

Reply:
Updated this.